### PR TITLE
Add "Use Thumbor" blog post

### LINKED ISSUE #1149: codelab: Use Thumbor to optimize images
**A one to two sentence description of your post**
Codelab on using Thumbor.

Author: @khempenius 
Reviewer: @mihajlija 

**What are you planning to teach folks in this post?**
How to optimize images with Thumbor.
Related blog post: "How to Install Thumbor" (https://github.com/GoogleChrome/web.dev/issues/1148)

**Target publish date:** `<2019-07-31>`

- [ ] Check this box if this is a hard deadline.

**Process**

- [x] 1. I've done my [proposal](https://docs.google.com/document/d/1cfFyi0EaCJF3iEjiYWffWVgrf9ruwMVzQxF0peT3y08/edit?usp=sharing)
- [x] 2. Proposal approved — _reviewer checks this_
- [x] 3. I've added a draft to my proposal doc
- [x] 4. Draft approved — _reviewer checks this_
- [x] 5. I've submitted a pull request #1490
- [ ] 6. Pull request approved — _reviewer checks this_

### PR DESCRIPTION
**Pull request by khempenius**
Fixes #1149

### CODE REVIEW

File: src/site/content/en/blog/use-thumbor/index.md [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+<figure class="w-figure  w-figure--center">
+  <img src="./Use-thumbor11.jpg" alt="DevTools screenshot showing the content-type (WebP) of an image" class="w-screenshot">
+</figure>
+
+
+## Next Steps
+
+Try out other [filters](https://thumbor.readthedocs.io/en/latest/filters.html) and transformations on the `hero.jpg` image.
+
+If you're following along using your own Thumbor installation, check out the appendix below that explains how and why to use the thumbor.conf file.
```
Comment (kaycebasques) [inline src/site/content/en/blog/use-thumbor/index.md]:
```suggestion
If you're following along using your own Thumbor installation, check out the appendix below that explains how and why to use the `thumbor.conf` file.
```

File: src/site/content/en/blog/use-thumbor/index.md [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+
+
+## Appendix: Thumbor.conf
```
Comment (kaycebasques) [inline src/site/content/en/blog/use-thumbor/index.md]:
```suggestion
## Appendix: `thumbor.conf`
```

File: src/site/content/en/blog/use-thumbor/index.md [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+
+Many of the configuration options discussed in this post, plus many others, can be established as defaults by setting up and using the `thumbor.conf` file. Settings in the `thumbor.conf` file will be applied to all images unless overridden by the URL string parameters.
```
Comment (kaycebasques) [inline src/site/content/en/blog/use-thumbor/index.md]:
```suggestion
Many of the configuration options discussed in this post, plus many others, can be established as defaults by setting up and using a `thumbor.conf` configuration file. Settings in the `thumbor.conf` file will be applied to all images unless overridden by the URL string parameters.
```

File: src/site/content/en/blog/use-thumbor/index.md [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+
+The best way to start setting up your `thumbor.conf` file is to use the `thumbor-config` command to create a new config.
```
Comment (kaycebasques) [inline src/site/content/en/blog/use-thumbor/index.md]:
```suggestion
1. Run the `thumbor-config` command to create a new `thumbor.conf` file.
```

File: src/site/content/en/blog/use-thumbor/index.md [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+
+```bash
+thumbor-config > ./thumbor.conf
+```
+
+The comments in this config will list and explain all Thumbor configurations options and their defaults. To change a setting, uncomment the line and change the text. The following are a few of the settings that you may find useful to set: `QUALITY`, `AUTO_WEBP`, `MAX_WIDTH` and `MAX_HEIGHT`, and `ALLOW_ANIMATED_GIFS`.
```
Comment (kaycebasques) [inline src/site/content/en/blog/use-thumbor/index.md]:
```suggestion
1. Open your new `thumbor.conf` file. The `thumbor-config` command listed and explained all Thumbor configuration options.
1. Configure settings by uncommenting lines and changing the default values.

You may find it useful to set the following settings:

* `QUALITY`
* `AUTO_WEBP`
* `MAX_WIDTH` and `MAX_HEIGHT`
* `ALLOW_ANIMATED_GIFS`
```

File: src/site/content/en/blog/use-thumbor/index.md [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+
+In order for the thumbor.conf settings to apply, Thumbor must be run using the `--conf` flag. This tells Thumbor the location of the thumbor.conf file.
```
Comment (kaycebasques) [inline src/site/content/en/blog/use-thumbor/index.md]:
```suggestion
1. Run Thumbor with the `--conf` flag in order to apply your `thumbor.conf` settings.
```

File: src/site/content/en/blog/use-thumbor/index.md [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+This part of the URL specifies the desired size of the output image. This can be omitted if you don't want to change the size of the image. Thumbor will use different approaches like cropping or scaling to achieve the desired size depending on the other URL parameters. The next section of this post explains how to resize images in more detail.
+
+Try it now:
+
+1. Click the URL below to view the image served at its original size. The image opens in a new tab.
+
+<a href="http://34.67.235.246:8888/unsafe/https://web.dev/backdrop-filter/hero.jpg" target="_blank" rel="noreferrer">http://34.67.235.246:8888/unsafe/https://web.dev/backdrop-filter/hero.jpg</a>
+
+<figure class="w-figure  w-figure--center">
+  <img src="./Use-thumbor1.jpg" alt="Image at original size" class="w-screenshot">
```
Comment (kaycebasques) [inline src/site/content/en/blog/use-thumbor/index.md]: Please add a `figcaption` for all images in the post

File: src/site/content/en/blog/use-thumbor/index.md [hunk 8]
```diff
@@ -0,0 +1,217 @@
...
+The `unsafe` part of the URL indicates that you're using Thumbor without a [security key](https://github.com/thumbor/thumbor/wiki/security). A security key prevents a user from making unauthorized changes to your image URLs. By changing the image URL, a user could use your server (and your hosting bill) to resize their images, or, more maliciously, to overload your server. This guide won't cover setting up Thumbor's security key feature.
+
+
+### Size
+
+This part of the URL specifies the desired size of the output image. This can be omitted if you don't want to change the size of the image. Thumbor will use different approaches like cropping or scaling to achieve the desired size depending on the other URL parameters. The next section of this post explains how to resize images in more detail.
+
+Try it now:
+
+1. Click the URL below to view the image served at its original size. The image opens in a new tab.
```
Comment (kaycebasques) [inline src/site/content/en/blog/use-thumbor/index.md]:
```suggestion
1. Click the following URL to view the image served at its original size in a new tab: <a href="http://34.67.235.246:8888/unsafe/https://web.dev/backdrop-filter/hero.jpg" target="_blank" rel="noreferrer">http://34.67.235.246:8888/unsafe/https://web.dev/backdrop-filter/hero.jpg</a>
```

Now that I see the rendered article, I realize it's a little awkward to have the URL below the instructions. Please use the format above for all the instructions. So each instruction should end with a colon, and then we list the URL to be clicked

File: src/site/content/en/blog/use-thumbor/index.md [hunk 9]
```diff
@@ -0,0 +1,217 @@
...
+The [format](https://thumbor.readthedocs.io/en/latest/format.html) filter converts images to `jpeg`, `webp`, `gif`, or `png`. Keep in mind that if you're optimizing for performance you should [use either JPEG or WebP](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/image-optimization) as PNG and GIF files tend to be significantly larger and do not compress as well.
+
+Try it now:
+
+1. Convert the image to WebP. If you open the **Network** panel of DevTools the document's **Content-Type response header** shows that the server returned a WebP image.
+
+<a href="http://34.67.235.246:8888/unsafe/filters:format(webp)/https://web.dev/backdrop-filter/hero.jpg" target="_blank" rel="noreferrer">http://34.67.235.246:8888/unsafe/filters:format(webp)/https://web.dev/backdrop-filter/hero.jpg</a>
+
+<figure class="w-figure  w-figure--center">
+  <img src="./Use-thumbor11.jpg" alt="DevTools screenshot showing the content-type (WebP) of an image" class="w-screenshot">
```
Comment (kaycebasques) [inline src/site/content/en/blog/use-thumbor/index.md]: Can you use this screenshot instead? The `content-type` text is hard to read. You'll need to add the red box around `content-type` again.

image

File: src/site/content/en/blog/use-thumbor/index.md [hunk 10]
```diff
@@ -0,0 +1,217 @@
...
+
+## Thumbor URL Format
+
+As mentioned in [Use Image CDNs to Optimize Images](https://web.dev/image-cdns), each image CDN uses a slightly different URL format for images. Figure 1 represents Thumbor's format.
+
+<figure class="w-figure  w-figure--center">
+  <img src="./Use-thumbor0.jpg" alt="A Thumbor URL has the following components: origin, security key, size, filters and image." class="w-screenshot">
+</figure>
+
+_Figure 1. Thumbor's URL format_
```
[the remark this answers is not in the document]
Comment (robdodson) [reply src/site/content/en/blog/use-thumbor/index.md]: one small note: I think we stopped doing the `Figure 1.` style of figcaptions a while back. I think you can omit that and just use "Thumbor's URL format".

[the remark this answers is not in the document]
Comment (robdodson) [reply src/site/content/en/blog/use-thumbor/index.md]: You can look at the image docs for examples. https://web.dev/handbook/markup-media these no longer use the `Figure X` style.

[the remark this answers is not in the document]
Comment (kaycebasques) [reply src/site/content/en/blog/use-thumbor/index.md]: Thanks, Rob. I thought I had seen it on other pages but `grep` tells me that I'm mistaken. Just a habit from developers.google.com/web.

File: src/site/content/en/blog/use-thumbor/index.md [hunk 11]
```diff
@@ -0,0 +1,211 @@
...
+1. Resize the image to a width of 200 pixels while maintaining original proportions: <a href="http://34.67.235.246:8888/unsafe/200x0/https://web.dev/backdrop-filter/hero.jpg" target="_blank" rel="noreferrer">http://34.67.235.246:8888/unsafe/200x0/https://web.dev/backdrop-filter/hero.jpg</a>
+
+
+<figure class="w-figure  w-figure--center">
+  <img src="./Use-thumbor7.jpg" alt="Image that is 200 pixels wide" class="w-screenshot">
+  <figcaption>Image resized to a width of 200 pixels</figcaption>
+</figure>
+
+
+2. Resize the image to a height of 500 pixels while maintaining original proportion<a href="http://34.67.235.246:8888/unsafe/0x500/https://web.dev/backdrop-filter/hero.jpg" target="_blank" rel="noreferrer">http://34.67.235.246:8888/unsafe/0x500/https://web.dev/backdrop-filter/hero.jpg</a>
```
Comment (kaycebasques) [inline src/site/content/en/blog/use-thumbor/index.md]:
```suggestion
2. Resize the image to a height of 500 pixels while maintaining original proportion: <a href="http://34.67.235.246:8888/unsafe/0x500/https://web.dev/backdrop-filter/hero.jpg" target="_blank" rel="noreferrer">http://34.67.235.246:8888/unsafe/0x500/https://web.dev/backdrop-filter/hero.jpg</a>
```

Just missing a colon and a space

File: src/site/content/en/blog/use-thumbor/index.md [hunk 12]
```diff
@@ -0,0 +1,211 @@
...
+
+The [format](https://thumbor.readthedocs.io/en/latest/format.html) filter converts images to `jpeg`, `webp`, `gif`, or `png`. Keep in mind that if you're optimizing for performance you should [use either JPEG or WebP](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/image-optimization) as PNG and GIF files tend to be significantly larger and do not compress as well.
+
+Try it now:
+
+1. Convert the image to WebP. If you open the **Network** panel of DevTools the document's **Content-Type response header** shows that the server returned a WebP image: <a href="http://34.67.235.246:8888/unsafe/filters:format(webp)/https://web.dev/backdrop-filter/hero.jpg" target="_blank" rel="noreferrer">http://34.67.235.246:8888/unsafe/filters:format(webp)/https://web.dev/backdrop-filter/hero.jpg</a>
+
+<figure class="w-figure  w-figure--center">
+  <img src="./Use-thumbor11.jpg" alt="DevTools screenshot showing the content-type (WebP) of an image" class="w-screenshot">
+  <figcaption>The `content-type` header in DevTools</figcaption>
```
Comment (kaycebasques) [inline src/site/content/en/blog/use-thumbor/index.md]:
```suggestion
  <figcaption>The <code>content-type</code> header in DevTools</figcaption>
```

Markdown doesn't work inside of HTML

File: src/site/content/en/blog/use-thumbor/index.md [hunk 13]
```diff
@@ -0,0 +1,211 @@
...
+### File Formats
+
+The [format](https://thumbor.readthedocs.io/en/latest/format.html) filter converts images to `jpeg`, `webp`, `gif`, or `png`. Keep in mind that if you're optimizing for performance you should [use either JPEG or WebP](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/image-optimization) as PNG and GIF files tend to be significantly larger and do not compress as well.
+
+Try it now:
+
+1. Convert the image to WebP. If you open the **Network** panel of DevTools the document's **Content-Type response header** shows that the server returned a WebP image: <a href="http://34.67.235.246:8888/unsafe/filters:format(webp)/https://web.dev/backdrop-filter/hero.jpg" target="_blank" rel="noreferrer">http://34.67.235.246:8888/unsafe/filters:format(webp)/https://web.dev/backdrop-filter/hero.jpg</a>
+
+<figure class="w-figure  w-figure--center">
+  <img src="./Use-thumbor11.jpg" alt="DevTools screenshot showing the content-type (WebP) of an image" class="w-screenshot">
```
Comment (kaycebasques) [inline src/site/content/en/blog/use-thumbor/index.md]: This image looks low-quality but I can fix it in a subsequent PR

cc @robdodson

File: src/site/content/en/blog/use-thumbor/index.md [hunk 14]
```diff
@@ -0,0 +1,211 @@
...
+```
+
+2. Open your new `thumbor.conf` file. The `thumbor-config` command generated a file that lists and explains all Thumbor configuration options.
+
+3. Configure settings by uncommenting lines and changing the default values.
+You may find it useful to set the following settings:
+* `QUALITY`
+* `AUTO_WEBP`
+* `MAX_WIDTH` and `MAX_HEIGHT`
+* `ALLOW_ANIMATED_GIFS`
```
Comment (kaycebasques) [inline src/site/content/en/blog/use-thumbor/index.md]: These bullet-list items are formatting weirdly but I believe I was the one that suggested changing it to a bulleted list so I'll also fix this in a subsequent PR

cc @robdodson

File: src/site/content/en/blog/use-thumbor/index.md (hunk 13 above)
Comment (khempenius) [reply src/site/content/en/blog/use-thumbor/index.md]: I updated this image. It's still not super high quality bc I was trying to keep the file size reasonable but it's definitely better than the previous version.

File: src/site/content/en/blog/use-thumbor/index.md (hunk 14 above)
Comment (khempenius) [reply src/site/content/en/blog/use-thumbor/index.md]: I figured out the fix for this so need for a separate PR.

File: src/site/content/en/blog/use-thumbor/index.md [hunk 15]
```diff
@@ -0,0 +1,212 @@
+---
+title: Optimize images with Thumbor
+subhead: Thumbor can be used for free to resize, compress, and transform images on-demand.
+authors:
+  - katiehempenius
+date: 2019-09-20
+hero: hero.jpg
```
Comment (robdodson) [inline src/site/content/en/blog/use-thumbor/index.md]: I think this is the same hero image that was used in the other thumbor post. What do you think about swapping it?

Comment (khempenius) [reply src/site/content/en/blog/use-thumbor/index.md]: Updated ✔️

**Comment by robdodson** [comment]
LGTM, thanks @khempenius!

**Pull request merged**